sklearn/tests/test_props.py

```diff
@@ -0,0 +1,729 @@
+    ):
+        trs.fit(X, y).transform(X, sample_weight=my_weights)
+
+
+def test_get_metadata_routing():
+    class TestDefaultsBadMethodName(_MetadataRequester):
+        __metadata_request__fit = {
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+            "my_param": RequestType.ERROR_IF_PASSED,
+        }
+        __metadata_request__score = {
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+            "my_param": True,
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+        }
+        # this will raise an error since we don't understand "other_method" as a method
+        __metadata_request__other_method = {"my_param": True}
+
+    class TestDefaults(_MetadataRequester):
+        __metadata_request__fit = {
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+        }
+        __metadata_request__score = {
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+            "my_param": True,
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+        }
+        __metadata_request__predict = {"my_param": True}
+
+    with pytest.raises(
+        AttributeError, match="'MetadataRequest' object has no attribute 'other_method'"
+    ):
+        TestDefaultsBadMethodName().get_metadata_routing()
+
+    expected = {
+        "score": {
+            "my_param": RequestType.REQUESTED,
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+        },
+        "fit": {
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+        },
+        "predict": {"my_param": RequestType.REQUESTED},
+    }
+    assert_request_equal(TestDefaults().get_metadata_routing(), expected)
+
+    est = TestDefaults().set_score_request(my_param="other_param")
+    expected = {
+        "score": {
+            "my_param": "other_param",
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+        },
+        "fit": {
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+        },
+        "predict": {"my_param": RequestType.REQUESTED},
+    }
+    assert_request_equal(est.get_metadata_routing(), expected)
+
+    est = TestDefaults().set_fit_request(sample_weight=True)
+    expected = {
+        "score": {
+            "my_param": RequestType.REQUESTED,
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.ERROR_IF_PASSED,
+        },
+        "fit": {
+            "my_other_param": RequestType.ERROR_IF_PASSED,
+            "sample_weight": RequestType.REQUESTED,
+        },
+        "predict": {"my_param": RequestType.REQUESTED},
+    }
+    assert_request_equal(est.get_metadata_routing(), expected)
+
+
+def test__get_default_requests():
```

Comment:
Is this meant to be testing MRO traversal? Can we try give it a name closer to the user story it's trying to test?